internal/pkg/checks/assert.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Ingress conformance test harness machinery
+package checks
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// AssertionSet performs checks and accumulates assertion errors
+type AssertionSet []error
+
+// Assert actual and expected parameters are deeply equal
+func (a *AssertionSet) Equals(actual interface{}, expected interface{}, errorTemplate string) {
+	if errorTemplate == "" {
+		errorTemplate = "Expected '%s' but was '%s'"
+	}
+	if !reflect.DeepEqual(expected, actual) {
+		err := fmt.Errorf(errorTemplate, expected, actual)
+		*a = append(*a, err)
+	}
+}
+
+// Assert the actual headers contains the expected headers key
+func (a *AssertionSet) ContainsHeaders(actual map[string][]string, expected []string, errorTemplate string) {
```

Comment:
errorTemplate is going be error prone. Can we go with a set "expected to  " etc and see what kinds of  specialization is  required?